### PR TITLE
Implementation of additional configs for clear-text reporter

### LINKED ISSUE #4209: Add additional config options to clear-text reporter
**Describe the solution you'd like**
Currently, the user, when using clear-text, does not have any options when it comes to the amount of information that is logged. Additional options (boolean controls) to control the three 'functions' (reportAllTests, reportAllMutants, ClearTextScoreTable) would give the user much more flexibility and control when using the clear-text reporter.

**Additional context**
Although I don't have the most amount of time on my hands, I did throw together a branch with the related changes included within it. It is very bare bones and does not dive into changing/implementing tests for this change of functionality, but I wanted to give a solid example of the new feature I am describing.  Link to the branch can be found: [here](https://github.com/C-Klug/stryker-js/commit/8c2a8ef96ea3e0e7465d9effcfa300f76fad91c1)

### PR DESCRIPTION
**Pull request by C-Klug**
Related issue: [here](https://github.com/stryker-mutator/stryker-js/issues/4209)

Implementation of additional configuration options for the clear-text reporter. These additions include the ability to enable/disable the three sections within the clear-text reporter that get printed: All tests, survived mutants, and score table.

Fixes #4209

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -85,7 +85,10 @@ Config file:
     "allowColor": true,
     "allowEmojis": false,
     "logTests": true,
-    "maxTestsToLog": 3
+    "maxTestsToLog": 3,
+    "reportAllTests": true,
+    "reportAllMutants": true,
```
Comment (nicojs) [inline docs/configuration.md:90]: Could we rephrase them as `reportTests` and `reportMutants`? That way we could in the future expand the options here to specify states as an array. For example `"reportMutants": ["killed"]`

File: docs/configuration.md [hunk 2]
```diff
@@ -439,6 +442,9 @@ The `clear-text` reporter supports three additional config options:
 - `allowColor` to use cyan and yellow in printing source file names and positions. This defaults to `true`, so specify as `clearTextReporter: { allowColor: false },` to disable if you must.
 - `logTests` to log the names of unit tests that were run to allow mutants. By default, only the first three are logged. The config for your config file is: `clearTextReporter: { logTests: true },`
 - `maxTestsToLog` to show more tests that were executed to kill a mutant when `logTests` is true. The config for your config file is: `clearTextReporter: { logTests: true, maxTestsToLog: 7 },`
+- `reportAllTests` to log all tests. This defaults to `true`, so specify as `clearTextReporter: { reportAllTests: false },` to disable if you must.
+- `reportAllMutants` to log all mutants. This defaults to `true`, so specify as `clearTextReporter: { reportAllMutants: false },` to disable if you must.
+- `reportScoreTable` to log score table. This defaults to `true`, so specify as `clearTextReporter: { reportScoreTable: false },` to disable if you must.
```
Comment (nicojs) [inline docs/configuration.md:447]: The defaults can already be viewed above. I also rephased it a bit so it is more clear what to expect (I hope).

```suggestion
- `reportTests` to log tests. When set to `true` this reporter will report a list of all tests with the amounts of mutants they killed (if any).
- `reportMutants` to log mutants. When set to `true` this reporter will report each `Survived` and `NoCoverage` mutant to the stdout, as well as log other mutant states on debug log level.
- `reportScoreTable` to log the score table. When set to `true` this reporter will end with a summary table with a row per mutated file.
```

**Review by nicojs — CHANGES_REQUESTED** [review]
Thanks, this looks great! Some minor remarks regarding the naming.

File: docs/configuration.md (hunk 1 above)
Comment (C-Klug) [reply docs/configuration.md]: Absolutely, those naming conventions do fit better especially for future-proofing for later expansion.

File: docs/configuration.md (hunk 2 above)
Comment (C-Klug) [reply docs/configuration.md]: Those descriptions look great, will update to them!

**Review by nicojs — APPROVED** [review]
Looks great! Will merge it just before our next minor release (next week probably)

File: packages/api/schema/stryker-core.json [hunk 3]
```diff
@@ -82,6 +82,21 @@
...
           "minimum": 0,
           "default": 3
+        },
+        "reportTests": {
+          "description": "Indicates whether or not to log all tests.",
+          "type": "boolean",
+          "default": true
+        },
+        "reportMutants": {
+          "description": "Indicates whether or not to log all mutants.",
```
Comment (danny12321) [inline packages/api/schema/stryker-core.json:92]:
```suggestion
          "description": "Indicates whether or not to log all (survived and no coverage) mutants.",
```

**Review by danny12321 — APPROVED** [review]

**Pull request merged**